shell/platform/fuchsia/flutter/tests/fake_session_unittests.cc

```diff
@@ -0,0 +1,459 @@
+                        ElementsAre(Pointee(IsEntityNode(
+                            nodes[3]->id(), node_label + std::string(1, '3'),
+                            IsEmpty())))))))));
+  }
+
+  // Destroy node #3.  It should still exist in the tree and the labels map
+  // because it has a grand-parent, but it is removed from the resource map.
+  nodes[3].reset();
+  session.Present2(0u, 0u, [](auto...) {});
+  loop().RunUntilIdle();
+  {
+    auto scene_graph = fake_session().SceneGraph();
+    EXPECT_EQ(scene_graph.root_view_id, kInvalidFakeResourceId);
+    EXPECT_EQ(scene_graph.buffer_collection_map.size(), 0u);
+    EXPECT_EQ(scene_graph.resource_map.size(), 1u);
+    EXPECT_EQ(scene_graph.labels_map.size(), 3u);
+    for (size_t i = 1; i < 4; i++) {
+      const std::string node_i_label = node_label + std::string(1, '0' + i);
+      ASSERT_EQ(scene_graph.labels_map.count(node_i_label), 1u);
+      ASSERT_EQ(scene_graph.resource_map.count(nodes[i]->id()),
+                i < 2 ? 1u : 0u);
+
+      const auto node_i_label_resources = scene_graph.labels_map[node_i_label];
+      EXPECT_EQ(node_i_label_resources.size(), 1u);
+      EXPECT_FALSE(node_i_label_resources[0].expired());
+
+      if (i < 2) {
+        const auto node_i = scene_graph.resource_map[nodes[i]->id()];
+        EXPECT_EQ(node_i_label_resources[0].lock(), node_i);
+      } else {
+        EXPECT_EQ(scene_graph.resource_map.count(nodes[i]->id()), 0u);
+      }
+    }
+
+    EXPECT_THAT(scene_graph.resource_map[nodes[1]->id()],
+                Pointee(IsEntityNode(
+                    nodes[1]->id(), node_label + std::string(1, '1'),
+                    ElementsAre(Pointee(IsEntityNode(
+                        nodes[2]->id(), node_label + std::string(1, '2'),
+                        ElementsAre(Pointee(IsEntityNode(
+                            nodes[3]->id(), node_label + std::string(1, '3'),
+                            IsEmpty())))))))));
+  }
+}
+
+TEST_F(FakeSessionTest, BasicSceneGraph) {
+  const std::string debug_label = GetCurrentTestName();
+  const std::string node_label = "ChildNode";
+  scenic::Session session = CreateSession();
+
+  // Create and present initial scene graph.  The `Present` hasn't been
+  // processed yet, so the session's view of the scene graph is still empty.
+  fuchsia::ui::views::ViewRef view_ref;
+  auto [view_token, view_holder_token] = scenic::ViewTokenPair::New();
+  auto view_ref_pair = scenic::ViewRefPair::New();
+  view_ref_pair.view_ref.Clone(&view_ref);
+  scenic::View root_view(&session, std::move(view_token),
+                         std::move(view_ref_pair.control_ref),
+                         std::move(view_ref_pair.view_ref), debug_label);
+  scenic::EntityNode child_node(&session);
+  child_node.SetLabel(node_label);
+  root_view.AddChild(child_node);
+  session.Present2(0u, 0u, [](auto...) {});
+  EXPECT_GE(fake_session().command_queue().size(), 0u);
```

Comment:
This examining the cmd queue on the server side, right?

Reply:
Yeah, only to check that it's not empty.  I changes this to EXPECT_GT